### PR TITLE
CORE-329 Convert App Details dialog in Belphegor to React

### PR DESCRIPTION
**Pull request by ash3rz**
image

### CODE REVIEW

File: react-components/src/apps/admin/AdminAppDetails.js [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+            setSubmitting(false);
+        });
+};
+
+function mapPropsToValues(props) {
+    const { app, documentation } = props;
+    return {
+        app: app,
+        documentation: documentation,
+    };
```
Comment (psarando) [inline react-components/src/apps/admin/AdminAppDetails.js]: Similar to destructuring, the key doesn't have to be specified here if it's the same as the variable name:
```javascript
     return {
        app,
        documentation,
    };
```
And to get really terse, we can do away with this function definition all together and use an `=>` function below:
```javascript
 export default withFormik({
    enableReinitialize: true,
    mapPropsToValues: ({ app, documentation }) => ({ app, documentation }),
    handleSubmit,
})(withI18N(AdminAppDetailsDialog, messages));
```

File: react-components/stories/apps/admin/AdminAppDetails.stories.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+class AdminAppDetailsTest extends Component {
+    render() {
+        const presenter = {
+            closeAppDetailsDlg: () => console.log("close"),
+
+            onSaveAppSelected: (app, resolve, reject) => {
+                setTimeout(() => {
+                    console.log("Save app selected", app);
+                    resolve();
```
Comment (psarando) [inline react-components/stories/apps/admin/AdminAppDetails.stories.js]: 👍

File: react-components/src/apps/admin/AdminAppDetails.js [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+                    onClick={handleSubmit}
+                >
+                    {getMessage("save")}
+                </Button>
+            </DialogActions>
+        </Dialog>
+    );
+}
+
+const StyledAdminAppDetailsFrom = withStyles(styles)(AdminAppDetailsForm);
```
[the remark this answers is not in the document]
Comment (ash3rz) [reply react-components/src/apps/admin/AdminAppDetails.js]: Oops!  Good catch

File: react-components/src/apps/admin/AdminAppDetails.js (hunk 1 above)
Comment (ash3rz) [reply react-components/src/apps/admin/AdminAppDetails.js]: Ah, terse is nicer!
I end up needing to modify this a bit in #438, so I'll save the update for that PR

**Comment by ash3rz** [comment]
Thanks for the review!

**Pull request merged**